unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/AssetManager/Common/AssetPromise.cs

```diff
@@ -70,9 +70,11 @@ protected void CallAndClearEvents(bool isSuccess = true)
             OnPreFinishEvent?.Invoke(this);
             OnPreFinishEvent = null;
 
-            Action<AssetType> finalEvent = isSuccess ? OnSuccessEvent : OnFailEvent;
-
-            finalEvent?.Invoke(asset);
+            if (raiseEvents)
```

Comment:
We already have `ClearEvents` for this case. I think we may be duplicating behaviour here.

Reply:
Yeap you al totally right. Reverted that change.